griptape/drivers/sql/snowflake_sql_driver.py

```diff
@@ -0,0 +1,85 @@
+from typing import Callable, Optional
+from griptape.drivers import BaseSqlDriver
+from attr import Factory, define, field
+from sqlalchemy import create_engine, text, MetaData, Table
+from sqlalchemy.engine import Engine
+from sqlalchemy.exc import NoSuchTableError
+from snowflake.connector import SnowflakeConnection
+
+
+@define
+class SnowflakeSqlDriver(BaseSqlDriver):
+    snowflake_connection_function: Callable[[], SnowflakeConnection] = field(
+        kw_only=True
+    )
+    engine: Engine = field(
+        default=Factory(
+            # Creator bypasses the URL param
+            # https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine.params.creator
+            lambda self: create_engine(
+                "snowflake://not@used/db", creator=self.snowflake_connection_function
+            ),
+            takes_self=True,
+        ),
+        kw_only=True,
+    )
+
+    @snowflake_connection_function.validator
+    def validate_snowflake_connection_function(
+        self, _, snowflake_connection_function: Callable[[], SnowflakeConnection]
+    ) -> None:
+        snowflake_connection = snowflake_connection_function()
+        if not isinstance(snowflake_connection, SnowflakeConnection):
+            raise ValueError(
+                "The snowflake_connection_function must return a SnowflakeConnection"
+            )
+        if not snowflake_connection.schema or not snowflake_connection.database:
```

Comment:
Is the schema name really always required in Snowflake?

Reply:
It seems that the sqlalchemy snowflake dialect requires the fully qualified object, so we need to require the schema and database name here.